src/Test/Tasty/Ingredients/Rerun.hs

```diff
@@ -91,13 +93,15 @@ import qualified Test.Tasty.Options as Tasty
 import qualified Test.Tasty.Runners as Tasty
 
 --------------------------------------------------------------------------------
-newtype RerunLogFile = RerunLogFile FilePath
+data RerunLogFile
+  = DefaultRerunLogFile
+  | CustomRerunLogFile FilePath
 
 instance Tasty.IsOption RerunLogFile where
   optionName = return "rerun-log-file"
-  optionHelp = return "Location of the log file (default: .tasty-rerun-log)"
-  defaultValue = RerunLogFile ".tasty-rerun-log"
-  parseValue = Just . RerunLogFile
+  optionHelp = return "Location of the log file (default: .tasty-rerun-log.EXECUTABLE, where EXECUTABLE is the base name of the calling program)"
+  defaultValue = DefaultRerunLogFile
```

Comment:
I'd probably prefer to keep `newtype RerunLogFile` and stuff as is, and do `defaultValue = unsafePerformIO $ ... getExecutablePath ...` Is it too evil?